### PR TITLE
add notebooks in the doc

### PR DESCRIPTION
**Pull request by videlec**
Use [myst-nb](https://pypi.org/project/myst-nb/) to have notebooks in the documentation (in markdown format)

Checklist
* [x] Added an entry in `doc/news/`
* [x] Added a test for this change
* [x] include @chriszhang3  notebook

### CODE REVIEW

**Comment by videlec** [comment]
@saraedum Do you have an idea why the manual build in CI fails with
```
AttributeError: module 'sage_docbuild.conf' has no attribute 'pygments_style'
```
?

**Comment by videlec** [comment]
I copied this snippet from sage-flatsurf in `conf.py`
```
# Imitate the look of the SageMath documentation.
html_theme = sage_docbuild.conf.html_theme
html_theme_options = sage_docbuild.conf.html_theme_options
pygments_style = sage_docbuild.conf.pygments_style
pygments_dark_style = sage_docbuild.conf.pygments_dark_style
html_css_files = sage_docbuild.conf.html_css_files
```

**Comment by saraedum** [comment]
> @saraedum Do you have an idea why the manual build in CI fails with
> 
> ```
> AttributeError: module 'sage_docbuild.conf' has no attribute 'pygments_style'
> ```
> 
> ?

I guess `furo` is missing here:

```py
if PythonModule("furo").is_present():
    pygments_style = "sphinx"
```

File: doc/source/conf.py [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+import sage_docbuild.conf
+
 # -- Project information -----------------------------------------------------
 
 project = 'surface-dynamics'
 copyright = '2021, the surface-dynamics authors'
 author = 'the surface-dynamics authors'
 
 # The full version, including alpha/beta/rc tags
-release = '0.4.7'
```
Comment (saraedum) [inline doc/source/conf.py:8]: This change is not needed. The version number is maintained by rever.

**Comment by videlec** [comment]
Thanks @saraedum ! Now there is another annoying error
```
/home/runner/work/surface-dynamics/surface-dynamics/doc/source/examples/interval_exchanges.md: Executing notebook using local CWD [mystnb]

Exception occurred:
  File "/usr/share/miniconda3/envs/test/lib/python3.10/site-packages/nbclient/client.py", line 1005, in async_execute_cell
    raise DeadKernelError("Kernel died") from None
nbclient.exceptions.DeadKernelError: Kernel died
The full traceback has been saved in /tmp/sphinx-err-v_zmamjl.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
```

**Comment by saraedum** [comment]
> Seems to be because myst_nb ignores the `#optional : XXX` directives that I copy/pasted...

I don't think it matters since all the dependencies are available anyway.

**Comment by saraedum** [comment]
We get a `no type named 'TStreamerInfo' in namespace 'CppyyLegacy'` followed by a segfault.

**Comment by saraedum** [comment]
> We get a `no type named 'TStreamerInfo' in namespace 'CppyyLegacy'` followed by a segfault.

It's fixed now :tada:

**Comment by saraedum** [comment]
Here's the logo I am using now:

logo

**Comment by videlec** [comment]
> Here's the logo I am using now:
> 
> logo

Nice. Though, we can not read the iet on the picture... I will think about a way to make it visible nicely.

**Pull request merged**